File: src/main/java/cloud/CreateDirServlet.java
package cloud;

import java.io.File;
import java.io.IOException;

import javax.servlet.ServletException;
import javax.servlet.annotation.WebServlet;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

import com.google.gson.Gson;

import cloud.bean.CloudFile;
import cloud.bean.CreateDir;
import cloud.bean.Response;

@WebServlet("/createdir")
public class CreateDirServlet extends HttpServlet {

	@Override
	protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {

		CreateDir params = JsonUtil.fromJsonReader(req.getReader(), CreateDir.class);

		if (params.paths.get(0).equals(Cons.Path.USER_DIR_STUB)) {
			params.paths.remove(0);
		}

		String path = FileUtil.getWholePath(Cons.Path.DATA_DIR, TokenUtil.getUsername(req.getParameter("token")),
				FileUtil.getWholePath(params.paths));

		boolean result = new File(path).mkdirs();

		System.out.print("CreateDirServlet: path->" + path + "\n");

		System.out.print("CreateDirServlet: mkdirs->" + new File(path).mkdirs() + "\n");

		resp.getWriter().write(JsonUtil.toJson(result
				? new Response<Void>(CodeMessage.OK.code, CodeMessage.OK.message, null)
				: new Response<Void>(CodeMessage.CREATE_DIR_FAIL.code, CodeMessage.CREATE_DIR_FAIL.message, null)));

	}
}

File: src/main/kotlin/cloud/servlet/CreateDirServlet.kt
package cloud.servlet

import cloud.model.net.CodeMessage
import cloud.model.net.CreateDir
import cloud.model.net.Response
import cloud.config.Cons
import cloud.manager.logger
import cloud.util.FileUtil
import cloud.util.CloudFileUtil
import cloud.util.JsonUtil
import cloud.util.TokenUtil
import java.io.File
import java.io.IOException
import jakarta.servlet.ServletException
import jakarta.servlet.annotation.WebServlet
import jakarta.servlet.http.HttpServlet
import jakarta.servlet.http.HttpServletRequest
import jakarta.servlet.http.HttpServletResponse

@WebServlet("/createdir")
class CreateDirServlet : HttpServlet() {
    @Throws(ServletException::class, IOException::class)
    override fun doPost(req: HttpServletRequest, resp: HttpServletResponse) {

        val paths = CloudFileUtil.getWholePath(JsonUtil.fromJsonReader(req.reader, CreateDir::class.java).paths, TokenUtil.getUsername(req.getParameter("token")))

        val path = FileUtil.getWholePath(Cons.Path.DATA_DIR, paths)

        val file = File(path)

        val response = if (file.exists()) {
            Response<Any?>(CodeMessage.DIR_ALREADY_EXIST.code, CodeMessage.DIR_ALREADY_EXIST.message, null)
        } else if (file.mkdirs()) {
            Response<Any?>(CodeMessage.OK.code, CodeMessage.OK.message, null)
        } else {
            Response<Void?>(CodeMessage.CREATE_DIR_FAIL.code, CodeMessage.CREATE_DIR_FAIL.message, null)
        }

        logger.info {
            "file:${file}"
        }

        resp.writer.write(JsonUtil.toJson(response))
    }
}